jamjar/jamjar/tasks/transcode_video.py

```diff
@@ -37,61 +61,100 @@ def do_upload_to_s3(self, s3_path, disk_path):
-                disk_path = os.path.join(out_dir, filename)
-                s3_path = os.path.join(s3_dir, filename)
+        for filename in os.listdir(local_dir):
+            extension = os.path.splitext(filename)[1].lower()
+            if extension in ['.mp4','.hls','.ts','.m3u8','.jpg']:
+                disk_path = os.path.join(local_dir, filename)
+                s3_path = os.path.join(settings.JAMJAR_ENV, str(self.video.uuid), filename)
                 self.do_upload_to_s3(s3_path, disk_path)
 
-    def delete_source(self, src_dir):
+    def delete_source(self):
         "deletes the source DIRECTORY on disk after uploading to s3"
-        if settings.VIDEOS_PATH in src_dir:
-            shutil.rmtree(src_dir)
-        else:
-            raise RuntimeError("trying to delete dir that shouldn't be deleted!: {:}".format(src_dir))
+        video_dir = self.video.get_video_dir()
+        shutil.rmtree(video_dir)
 
-    def fingerprint(self, src_filepath, video_id):
+    def fingerprint(self):
         "fingerprints an mp4 and inserts the fingerprints into the db"
-        lilo = Lilo(settings.LILO_CONFIG, src_filepath, video_id)
+        video_path = self.video.get_video_filepath('mp4')
+        lilo = Lilo(settings.LILO_CONFIG, video_path, self.video.id)
         matched_videos = lilo.recognize_track()
 
         if matched_videos is not None:
             for match in matched_videos:
-                Edge.new(video_id, match['video_id'], match['offset_seconds'], match['confidence'])
-
-        lilo.fingerprint_song()
+                Edge.objects.create(video1_id=self.video.id,video2_id=match['video_id'],offset=match['offset_seconds'],confidence=match['confidence'])
+
+        # Add this videos fingerprints to the Lilo DB
+        data = lilo.fingerprint_song()
+
+        if not data:
+            raise RuntimeError('Video already fingerprinted: {} - {}'.format(self.video.name, self.video.uuid))
+
+        return data.get('song_length')
+
+    def extract_thumbnail(self, video_length):
+        """
+        extracts a thumbnail from a video and uploads it to s3
+        Use the following avconv settings
+            -i - input video file
+            -vsync 1 - video sync method. only write one frame
+            -r 1 - writing frame rate.  Use 1 so we only get 1 frame.
+            -an - disable audio recording
+            -t 1 - stop writing after 1 second
+            -ss time - grab the frame at <time> seconds
+            -y output - output file
+        """
+        src = self.video.get_video_filepath('mp4')
+        out = self.video.get_video_filepath('jpg',filename='thumb')
+        try:
+            with open(os.devnull, "w") as devnull:
+                thumbnail_time = video_length/2.0
+                logger.info("Here")
+                subprocess.check_call(['avconv', '-i', src, '-vsync', '1', '-r', '1', '-an', '-t', '1', '-ss', str(thumbnail_time), '-y', out], stdout=devnull, stderr=devnull)
```

Comment:
this totally works but the pic is way too big -- it's more like a toenail than a thumbnail :)